src/utils/components/DynamicSSHKeyInjection/DynamicSSHKeyInjection.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { FC, useState } from 'react';
+
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { getIsDynamicSSHInjectionEnabled } from '@kubevirt-utils/resources/vm';
+import { Switch } from '@patternfly/react-core';
+
+type DynamicSSHKeyInjectionProps = {
+  isDisabled: boolean;
+  onSubmit: (checked: boolean) => void;
+  vm?: V1VirtualMachine;
+};
+export const DynamicSSHKeyInjection: FC<DynamicSSHKeyInjectionProps> = ({
+  isDisabled,
+  onSubmit,
+  vm,
+}) => {
+  const isEnabled = getIsDynamicSSHInjectionEnabled(vm);
+  const [isChecked, setIsChecked] = useState<boolean>(isEnabled || false);
```

Comment:
getIsDynamicSSHInjectionEnabled is returning a boolean, so why we need || false here?
try to see if  `const [isChecked, setIsChecked] = useState<boolean>(getIsDynamicSSHInjectionEnabled(vm));` works for you as you also have isDisabled and isEnabled here which is confusing